### PR TITLE
refactor: consume shared audit as published dep in portal-backend

### PR DESCRIPTION
**Pull request by mushrafmim**
## Summary

Migrates `portal-backend` off the local `replace`-based audit module (`github.com/gov-dx-sandbox/shared/audit`) to the published `github.com/OpenDIF/opendif-core/shared/audit`, and folds the nested `shared/utils` module into the main module. This aligns portal-backend with the other services (consent-engine, orchestration-engine) that already consume shared modules from the module proxy, removing local `replace` directives.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [x] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Other (please describe):

## Changes Made

- **`go.mod` / `go.sum`**: Removed the `replace` directives for `shared/audit`, `portal-backend/shared/utils`, and `portal-backend/models`; dropped the local `shared/utils` require. Added `github.com/OpenDIF/opendif-core/shared/audit` as a direct dependency (via the module proxy) and tidied.
- **`portal-backend/shared/utils/go.mod`**: Deleted — this package is now part of the main module rather than a nested module.
- **`main.go`**: Updated the audit import to `github.com/OpenDIF/opendif-core/shared/audit` and the `audit.NewClient` / `audit.InitializeGlobalAudit` references.
- **`v1/middleware/audit_middleware.go`**: Updated import; switched the `LogAudit` client type to `audit.Auditor`. Kept the `client == nil` guard so a nil audit client is skipped safely (the shared package documents nil as a supported "audit disabled" state) rather than panicking on `IsEnabled()`.
- **`v1/middleware/audit_middleware_test.go`**: Updated import path and type references to the published package.

## Testing

- [x] I have tested this change locally
- [ ] I have added unit tests for new functionality
- [x] I have tested edge cases
- [x] All existing tests pass

`go build ./...`, `go vet ./...`, and `go test ./...` all pass. The nil-client path is covered by the existing `v1/middleware` tests. `go mod tidy` is stable (no drift).

## Checklist

- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have checked that there are no merge conflicts

## Additional Notes

No functional change to audit behavior — this is a dependency-sourcing refactor. The `client == nil || !client.IsEnabled()` guard in `LogAudit` is intentionally retained: the published package documents a nil client as a valid "audit disabled" case, so dropping the nil check would risk a panic.

### CODE REVIEW

File: portal-backend/v1/middleware/audit_middleware.go [hunk 1]
```diff
@@ -5,13 +5,13 @@ import (
...
 	"github.com/gov-dx-sandbox/portal-backend/v1/models"
-	auditpkg "github.com/gov-dx-sandbox/shared/audit"
 )
 
 // LogAudit logs an audit event for portal-backend operations by extracting request info and creating an audit log
-func LogAudit(client auditpkg.AuditClient, r *http.Request, resource string, resourceID *string, status string) {
-	// Skip if audit client is not enabled
+func LogAudit(client audit.Auditor, r *http.Request, resource string, resourceID *string, status string) {
+	// Skip if audit client is not set or not enabled
 	if client == nil || !client.IsEnabled() {
```
[the remark this answers is not in the document]
Comment (mushrafmim) [reply portal-backend/v1/middleware/audit_middleware.go:15]: Addressed. This issue is not reachable.

File: portal-backend/v1/middleware/audit_middleware.go [hunk 2]
```diff
@@ -5,13 +5,13 @@ import (
...
 
+	"github.com/OpenDIF/opendif-core/shared/audit"
 	"github.com/gov-dx-sandbox/portal-backend/v1/models"
-	auditpkg "github.com/gov-dx-sandbox/shared/audit"
 )
 
 // LogAudit logs an audit event for portal-backend operations by extracting request info and creating an audit log
-func LogAudit(client auditpkg.AuditClient, r *http.Request, resource string, resourceID *string, status string) {
-	// Skip if audit client is not enabled
+func LogAudit(client audit.Auditor, r *http.Request, resource string, resourceID *string, status string) {
```
[the remark this answers is not in the document]
Comment (mushrafmim) [reply portal-backend/v1/middleware/audit_middleware.go:13]: Addressed

**Review by sthanikan2000 — APPROVED** [review]

File: portal-backend/v1/middleware/audit_middleware_test.go [hunk 3]
```diff
@@ -133,20 +133,20 @@ func TestAuditMiddleware_ThreadSafety(t *testing.T) {
 				url = "" // Mix enabled and disabled instances
 			}
 
-			var client auditpkg.AuditClient
+			var client audit.AuditClient
```
[the remark this answers is not in the document]
Comment (mushrafmim) [reply portal-backend/v1/middleware/audit_middleware_test.go:136]: Addressed

File: portal-backend/v1/middleware/audit_middleware_test.go [hunk 4]
```diff
@@ -9,28 +9,28 @@ import (
...
 	"time"
 
+	"github.com/OpenDIF/opendif-core/shared/audit"
 	"github.com/gov-dx-sandbox/portal-backend/v1/models"
 	"github.com/gov-dx-sandbox/portal-backend/v1/utils"
-	auditpkg "github.com/gov-dx-sandbox/shared/audit"
 )
 
-// mockAuditClient implements auditpkg.AuditClient interface for testing
+// mockAuditClient implements audit.AuditClient interface for testing
```
[the remark this answers is not in the document]
Comment (mushrafmim) [reply portal-backend/v1/middleware/audit_middleware_test.go:17]: Addressed
